code/modules/paperwork/ticketmachine.dm

```diff
@@ -0,0 +1,201 @@
+			to_chat(user, "<span class='notice'>You insert [I] into [src] as it whirs nondescriptly.</span>")
+			qdel(I)
+			ticket_number = 0
+			current_number = 0
+			if(tickets.len)
+				for(var/obj/item/ticket_machine_ticket/ticket in tickets)
+					ticket.audible_message("<span class='notice'>\the [ticket] disperses!</span>")
+					qdel(ticket)
+				tickets.Cut()
+			max_number = initial(max_number)
+			update_icon()
+			return
+
+/obj/machinery/ticket_machine/proc/reset_cooldown()
+	ready = TRUE
+
+/obj/machinery/ticket_machine/attack_hand(mob/living/carbon/user)
+	. = ..()
+	if(!ready)
+		to_chat(user,"<span class='warning'>You press the button, but nothing happens...</span>")
+		return
+	if(ticket_number >= max_number)
+		to_chat(user,"<span class='warning'>Ticket supply depleted, please refill this unit with a hand labeller refill cartridge!</span>")
+		return
+	if((user in ticket_holders) && !(emagged))
+		to_chat(user, "<span class='warning'>You already have a ticket!</span>")
+		return
+	playsound(src, 'sound/machines/terminal_insert_disc.ogg', 100, FALSE)
+	ticket_number ++
+	to_chat(user, "<span class='notice'>You take a ticket from [src], looks like you're ticket number #[ticket_number]...</span>")
+	var/obj/item/ticket_machine_ticket/theirticket = new /obj/item/ticket_machine_ticket(get_turf(src))
+	theirticket.name = "Ticket #[ticket_number]"
+	theirticket.maptext = "<font color='#000000'>[ticket_number]</font>"
+	theirticket.saved_maptext = "<font color='#000000'>[ticket_number]</font>"
+	theirticket.ticket_number = ticket_number
+	theirticket.source = src
+	theirticket.owner = user
+	user.put_in_hands(theirticket)
+	ticket_holders += user
+	tickets += theirticket
+	if(emagged) //Emag the machine to destroy the HOP's life.
+		ready = FALSE
+		addtimer(CALLBACK(src, .proc/reset_cooldown), cooldown)//Small cooldown to prevent piles of flaming tickets
+		theirticket.fire_act()
+		user.drop_item(theirticket)
+		user.adjust_fire_stacks(1)
+		user.IgniteMob()
+		return
+
+/obj/item/ticket_machine_ticket
+	name = "Ticket"
+	desc = "A ticket which shows your place in the Head of Personnel's line. Made from Nanotrasen patented NanoPaper�. Though solid, its form seems to shimmer slightly. Feels (and burns) just like the real thing."
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "ticket"
+	maptext_x = 7
+	maptext_y = 10
+	w_class = WEIGHT_CLASS_TINY
+	resistance_flags = FLAMMABLE
+	max_integrity = 50
+	var/saved_maptext = null
+	var/mob/living/carbon/owner
+	var/obj/machinery/ticket_machine/source
+	var/ticket_number
+
+/obj/item/ticket_machine_ticket/attack_hand(mob/user)
+	. = ..()
+	maptext = saved_maptext //For some reason, storage code removes all maptext off objs, this stops its number from being wiped off when taken out of storage.
+
+/obj/item/ticket_machine_ticket/attackby(obj/item/P, mob/living/carbon/human/user, params) //Stolen from papercode
+	..()
+	if(is_hot(P))
+		if(HAS_TRAIT(user, CLUMSY) && prob(10))
```

Comment:
we don't have this trait. It's a mutation, specifically--we haven't concerted it.

Reply:
Fixed